website/static/js/nodesDelete.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * Controller for deleting a node and it childs (if they exists)
+ */
+'use strict';
+
+var $ = require('jquery');
+var $3 = window.$3;
+var ko = require('knockout');
+var Raven = require('raven-js');
+var $osf = require('./osfHelpers');
+var osfHelpers = require('js/osfHelpers');
+var m = require('mithril');
+var NodesDeleteTreebeard = require('js/nodesDeleteTreebeard');
+
+function _flattenNodeTree(nodeTree) {
+    var ret = [];
+    var stack = [nodeTree];
+    while (stack.length) {
+        var node = stack.pop();
+        ret.push(node);
+        stack = stack.concat(node.children);
+    }
+    return ret;
+}
+
+/**
+ * take treebeard tree structure of nodes and get a dictionary of parent node and all its
+ * children
+ */
+function getNodesOriginal(nodeTree, nodesOriginal) {
+    var flatNodes = _flattenNodeTree(nodeTree);
+    $.each(flatNodes, function(_, nodeMeta) {
+        nodesOriginal[nodeMeta.node.id] = {
+            id: nodeMeta.node.id,
+            title: nodeMeta.node.title,
+            isAdmin: nodeMeta.node.is_admin,
+            changed: false
+        };
+    });
+    nodesOriginal[nodeTree.node.id].isRoot = true;
+    return nodesOriginal;
+}
+
+/**
+ * patches all the nodes in a changed state
+ * uses API v2 bulk requests
+ */
+function patchNodesDelete(nodes) {
+    var nodesV2Url = window.contextVars.apiV2Prefix + 'nodes/';
+    var nodesPatch = $.map(nodes, function (node) {
+        return {
+            'type': 'nodes',
+            'id': node.id,
+            'attributes': {
+                'public': node.public
```

Comment:
So I'm guessing you copied this from the bulk privacy changes. A delete request doesn't need this hefty of a payload.

Reply:
Right.... I simply changed the method to DELETE without checking payload.